api/src/utils/csv-utils/csv-config-validation.ts

```diff
@@ -0,0 +1,179 @@
+        throw new Error(`Duplicate header in CSV config: ${uppercasedHeader}`);
+      }
+
+      headerMap.set(uppercasedHeader, { ...headerConfig, staticHeader });
+    }
+  }
+
+  return headerMap;
+};
+
+/**
+ * Iterate over each cell in the CSV worksheet.
+ *
+ * @param {WorkSheet} worksheet - The worksheet
+ * @param {CSVConfig} config - The CSV configuration
+ * @param {(params: CSVParams, staticHeaderConfig?: CSVHeaderConfig & { staticHeaderConfig: string }) => void} callback - The callback function
+ * @returns {void}
+ */
+export const _forEachCSVCell = (
+  worksheet: WorkSheet,
+  config: CSVConfig,
+  callback: (params: CSVParams, staticHeaderConfig?: CSVHeaderConfig & { staticHeader: string }) => void
+): void => {
+  const configMap = _getCSVConfigMap(config);
+  const worksheetRows = getWorksheetRowObjects(worksheet);
+
+  for (let i = 0; i < worksheetRows.length; i++) {
+    const worksheetRow = worksheetRows[i];
+
+    for (const header in worksheetRow) {
+      const staticHeaderConfig = configMap.get(header);
+      const cell = worksheetRow[header];
+      const params = { cell, header, row: worksheetRow, rowIndex: i + 1 };
+
+      callback(params, staticHeaderConfig);
+    }
+  }
+};
+
+/**
+ * Validate the CSV static and dynamic headers against the CSV config.
+ *
+ * @param {WorkSheet} worksheet - The worksheet
+ * @param {CSVConfig} config - The CSV configuration
+ * @returns {CSVError[]} - The CSV errors
+ */
+export const validateCSVHeaders = (worksheet: WorkSheet, config: CSVConfig): CSVError[] => {
+  const csvErrors: CSVError[] = [];
+
+  const configUtils = new CSVConfigUtils(worksheet, config);
+
+  if (!configUtils.headers.length) {
+    return [{ rowIndex: 0, error: 'CSV is empty', solution: 'Add headers and data to CSV' }];
+  }
+
+  for (const [staticHeader, headerConfig] of Object.entries(config.staticHeadersConfig)) {
+    const worksheetHasStaticHeader = configUtils.staticHeaders.some((header) =>
+      [staticHeader, ...headerConfig.aliases].includes(header)
+    );
+
+    if (!worksheetHasStaticHeader) {
+      csvErrors.push({
+        rowIndex: 0,
+        error: 'CSV missing required header',
+        header: staticHeader,
+        solution: `Add header '${staticHeader}' to CSV`
+      });
+    }
+  }
+
+  if (!config.ignoreDynamicHeaders && !config.dynamicHeadersConfig && configUtils.dynamicHeaders.length) {
+    for (const unknownHeader of configUtils.dynamicHeaders) {
+      csvErrors.push({
+        rowIndex: 0,
+        error: 'Unknown header in CSV',
+        header: unknownHeader,
+        solution: `Remove header '${unknownHeader}' from CSV`
+      });
+    }
+  }
+
+  return csvErrors;
+};
+
+/**
+ * Validate the CSV cells against the CSV config.
+ *
+ * @param {WorkSheet} worksheet - The worksheet
+ * @param {CSVConfig} config - The CSV configuration
+ * @returns {CSVError[]} - The CSV errors
+ */
+export const validateCSVCells = (worksheet: WorkSheet, config: CSVConfig): CSVError[] => {
```

Comment:
nit: should this be prefixed with an underscore?

Same with the `validateCSVCells` function below.